pkgs/misc/dxvk/darwin-moltenvk-compat.patch

```diff
@@ -0,0 +1,59 @@
+Source: https://github.com/Gcenx/MoltenVK/commit/1f268bd45f6c928784d7cb90dedce382895f10ae
```

Comment:
Wouldn't it be better to expose a second MoltenVK in Nixpkgs that has this patch and simply use that?

Reply:
When I submitted MoltenVK to nixpkgs, I included a flag that would apply the patches, so you could have a patched or unpatched version as necessary. The feedback I got was to remove the patches and provide them elsewhere. It was suggested I do it as part of Wine, but doing it this way lets one use the DLLs with e.g., CrossOver.

Reply:
> @reckenrode nobody should ever use the custom MoltenVK/DXVK with CrossOver as that would invalidate all support tickets.

They shouldn’t (and they definitely shouldn’t open support tickets if they do), but people are going to do it as CrossOver lags behind upstream Wine. Ideally, WoW64 would be good enough to use without needing CrossOver, but it’s not there yet.

Reply:
> They should be included in the MoltenVK derivation but not in the package. If it was `moltenvk-patched` or something, I don't think anybody would object that.

Are there any precedents for that with other packages? I did a grep of nixpkgs but couldn’t find anything really. This patched MoltenVK is _only_ useful to DXVK. It should never be used for anything else.

> MoltenVK patches should go to MoltenVK but not enabled by default. Though it'd again make sense to simply use different trees if someone's maintaining patches there already. This way of providing patched versions would also make it easier to offer a dxvk-async package.

Are there best practices for providing packages that offer variants or require patched dependencies to work? I know `vscode-lldb` does the same as I’m doing here with DXVK (except it uses `overrideAttrs` to replace the `src` completely rather than apply patches). Are there others?

Reply:
> > @reckenrode nobody should ever use the custom MoltenVK/DXVK with CrossOver as that would invalidate all support tickets.
> 
> They shouldn’t (and they definitely shouldn’t open support tickets if they do), but people are going to do it as CrossOver lags behind upstream Wine. Ideally, WoW64 would be good enough to use without needing CrossOver, but it’s not there yet.

CrossOver has always lagged behind upstream wine but with good reason. But you’d be surprised at the number of users who actually follow _guides_ then open invalid bug reports.

Hopefully we’ll have everything in place to push the Winehq macOS packages directly to Winehq by the time the WoW64 work is complete. As Gijs and myself (mostly me) end up spending countless hours simply fixing issues compiling libraries for i386, Gijs ends up spending most of there free time running regression tests.

Reply:
Also, as far as the async-dxvk patches go, I’m not really sure how to handle those. From what I understand, they don’t always work for everyone and may not work for every game.

Reply:
> Also, as far as the async-dxvk patches go, I’m not really sure how to handle those. From what I understand, they don’t always work for everyone and may not work for every game.

Async can also get users banned

Reply:
> > Also, as far as the async-dxvk patches go, I’m not really sure how to handle those. From what I understand, they don’t always work for everyone and may not work for every game.
> 
> Async can also get users banned

Is that because it trips anti-cheat detection in some online games? I saw some old reddit posts re: _Overwatch_. Is there a better alternative to avoiding the hitching?

Reply:
I should add that I would be against including something in nixpkgs that could potentially affect users adversely, so I’m not sure async-dxvk would be a good idea. It’s for that same reason I wouldn’t want to submit a package for ffxiv that supported mods. Even if Square Enix has permissive as long as you don’t talk about them in game, I wouldn’t want to put users at risk in case they changed their minds.

Reply:
> CrossOver has always lagged behind upstream wine but with good reason. But you’d be surprised at the number of users who actually follow _guides_ then open invalid bug reports.

I’d hope that nixpkgs users’ first recourse would be either Matrix or opening a PR when something doesn’t work. It’s the same as any packaged software. So maybe that’s wishful thinking. 😬

> Hopefully we’ll have everything in place to push the Winehq macOS packages directly to Winehq by the time the WoW64 work is complete. As Gijs and myself (mostly me) end up spending countless hours simply fixing issues compiling libraries for i386, Gijs ends up spending most of there free time running regression tests.

Really looking forward to it. It would be nice to have parity between Darwin and Linux in nixpkgs. Even if you’re on a 32-bit capable version of macOS, nixpkgs just does not have the infrastructure in place to build the 32-bit libraries. i686-darwin is not supported at all in any way.